src/sources/journald.rs

```diff
@@ -83,11 +83,13 @@ impl SourceConfig for JournaldConfig {
 
         let include_units = match (self.units.is_empty(), self.include_units.is_empty()) {
             (false, false) => return Err(BuildError::BothUnitsAndIncludeUnits.into()),
-            (true, false) => {
+            (false, true) => {
                 warn!("The `units` setting is deprecated, use `include_units` instead");
                 &self.units
             }
-            (_, true) => &self.include_units,
+            // include_units is either empty or not, both cases are fine since units is
+            // certainly empty.
+            _ => &self.include_units,
```

Comment:
I don't see the point in changing this from a more explicit pattern to a catch-all. It doesn't improve clarity (as demonstrated by the need for a comment), nor would I expect it to improve code generation.

Reply:
True